### PR TITLE
[patreon] fix not being able to download embedded videos

### LINKED ISSUE #8973: Patreon embedded video not found anymore
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting that yt-dlp is broken on a **supported** site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

Worldwide

### Provide a description that is worded well enough to be understood

Today it seems that Patreon changed their embedding or embedded videos. The program fails on all embedded videos and creators I got access to. So it looks like a site change.

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
[debug] Command-line config: ['-vU', '--cookies-from-browser', 'firefox', 'https://www.patreon.com/posts/fairy-tail-30-85954582']
Extracting cookies from firefox
[debug] Extracting cookies from: "C:\Users\Black\AppData\Roaming\Mozilla\Firefox\Profiles\i7jxqj1d.default-release-1663947917890\cookies.sqlite"
Extracted 2322 cookies from firefox
[debug] Encodings: locale cp932, fs utf-8, pref cp932, out utf-8, error utf-8, screen utf-8
[debug] yt-dlp version nightly@2024.01.09.232723 from yt-dlp/yt-dlp-nightly-builds [95e82347b] (win_exe)
[debug] Python 3.8.10 (CPython AMD64 64bit) - Windows-10-10.0.19045-SP0 (OpenSSL 1.1.1k  25 Mar 2021)
[debug] exe versions: ffmpeg N-109468-gd39b34123d-20221230 (setts), ffprobe N-109468-gd39b34123d-20221230
[debug] Optional libraries: Cryptodome-3.19.1, brotli-1.1.0, certifi-2023.11.17, mutagen-1.47.0, requests-2.31.0, sqlite3-3.35.5, urllib3-2.1.0, websockets-12.0
[debug] Proxy map: {}
[debug] Request Handlers: urllib, requests, websockets
[debug] Loaded 1798 extractors
[debug] Fetching release info: https://api.github.com/repos/yt-dlp/yt-dlp-nightly-builds/releases/latest
Latest version: nightly@2024.01.09.232723 from yt-dlp/yt-dlp-nightly-builds
yt-dlp is up to date (nightly@2024.01.09.232723 from yt-dlp/yt-dlp-nightly-builds)
[Patreon] Extracting URL: https://www.patreon.com/posts/fairy-tail-30-85954582
[Patreon] 85954582: Downloading API JSON
ERROR: [Patreon] 85954582: No supported media found in this post
  File "yt_dlp\extractor\common.py", line 718, in extract
  File "yt_dlp\extractor\patreon.py", line 289, in _real_extract
  File "yt_dlp\extractor\common.py", line 1196, in raise_no_formats
```

### PR DESCRIPTION
**Pull request by johnvictorfs**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

It seems the `post_file['name']` attribute no longer exists in the Patreon video attributes (at least in the posts I tested, but unsure if it may exist in some cases), so the check for the case of m3u8 downloads was being skipped, so I changed it to also check for `post_file['duration']` and assume it is a video in this case. Is this a good enough check for this case?

296526586-797a58c6-98c5-49d8-9985-c956aaee63df

Fixes #8973

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check all of the following options that apply:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

</details>

### CODE REVIEW

**Comment by pukkandan** [comment]
What about the other code path in L272?

**Comment by johnvictorfs** [comment]
> What about the other code path in L272?

I haven't actually found any cases where `post_file['name']` is returned, so I'm not sure if that path would ever be executed. I initially didn't want to remove it, as Patreon accepts various methods for uploading videos, and I wasn't sure if there was a case where it would return a name.

**Comment by kevo885** [comment]
Sorry if this is dumb question but how do you use this 'fix". I am already on latest version of nightly

File: yt_dlp/extractor/patreon.py [hunk 1]
```diff
@@ -275,7 +275,7 @@ def _real_extract(self, url):
                     'ext': ext,
                     'url': post_file['url'],
                 }
-            elif name == 'video':
+            elif name == 'video' or post_file.get('duration'):
```
Comment (bashonly) [inline yt_dlp/extractor/patreon.py:278]:
```suggestion
            elif name == 'video' or determine_ext(post_file.get('url')) == 'm3u8':
```

**Review by bashonly — CHANGES_REQUESTED** [review]

**Comment by bashonly** [comment]
@kevo885 `yt-dlp --update-to master`

**Pull request merged**